ci/scripts/cpp_build.sh

```diff
@@ -181,6 +181,11 @@ if [ "${ARROW_USE_CCACHE}" == "ON" ]; then
     ccache -s
 fi
 
+if command -v sccache &> /dev/null; then
+  echo "=== sccache stats after the build ==="
+  sccache --show-stats
```

Comment:
@assignUser This causes an error on non-amd64 environment: https://app.travis-ci.com/github/apache/arrow/jobs/583028972#L3378

```text
+ sccache --show-stats
/arrow/ci/scripts/cpp_build.sh: line 186: /usr/local/bin/sccache: cannot execute binary file: Exec format error
ERROR: 126
```

Could you fix this as a follow-up task?